modules/snooker/src/main/ts/ephox/snooker/resize/Sizes.ts

```diff
@@ -130,19 +124,9 @@ export const getHeight = function (cell: SugarElement) {
   return get(cell, 'rowspan', getTotalHeight);
 };
 
-export const getGenericWidth = function (cell: SugarElement): Optional<GenericWidth> {
+export const getGenericWidth = function (cell: SugarElement): Optional<Dimension.Dimension<'fixed' | 'relative' | 'empty'>> {
   const width = getRawWidth(cell);
-  return width.bind(function (w) {
-    const match = rGenericSizeRegex.exec(w);
-    if (match !== null) {
-      return Optional.some({
-        width: parseFloat(match[1]),
-        unit: match[3]
-      });
-    } else {
-      return Optional.none<GenericWidth>();
-    }
-  });
+  return width.bind((w) => Dimension.parse(w, [ 'fixed', 'relative', 'empty' ]));
```

Comment:
The old regex allowed for unitless widths, so the idea was to keep it the same.

Reply:
Sounds like something we should probably change, but seeing as changing that is risky I'm okay for it to be left as is.

Reply:
I checked and we do want to allow unitless values here, as it'll fall back to getting the size from the `width` attribute in `getRawWidth` and that will just be a plain number such as `100`. That being said though, we probably should be adding `px` to that when getting the value from the attribute in `getRawWidth` since that's what it represents.